tests/test_packages/test_skills/test_weather.py

```diff
@@ -20,251 +20,56 @@
-
-        result = self.runner.invoke(
-            cli, [*CLI_LOG_OPTION, "install"], standalone_mode=False
-        )
-        assert result.exit_code == 0
-
-        try:
-            os.chdir(agent_one_dir_path)
-            process_one = subprocess.Popen(  # nosec
-                [
-                    sys.executable,
-                    "-m",
-                    "aea.cli",
-                    "run",
-                    "--connections",
-                    "fetchai/oef:0.2.0",
-                ],
-                stdout=subprocess.PIPE,
-                env=os.environ.copy(),
-            )
-            os.chdir(agent_two_dir_path)
-            process_two = subprocess.Popen(  # nosec
-                [
-                    sys.executable,
-                    "-m",
-                    "aea.cli",
-                    "run",
-                    "--connections",
-                    "fetchai/oef:0.2.0",
-                ],
-                stdout=subprocess.PIPE,
-                env=os.environ.copy(),
-            )
-
-            # TODO increase timeout so we are sure they work until the end of negotiation.
-            time.sleep(5.0)
-            process_one.send_signal(signal.SIGINT)
-            process_one.wait(timeout=10)
-            process_two.send_signal(signal.SIGINT)
-            process_two.wait(timeout=10)
-
-            assert process_one.returncode == 0
-            assert process_two.returncode == 0
-        finally:
-            poll_one = process_one.poll()
-            if poll_one is None:
-                process_one.terminate()
-                process_one.wait(2)
-
-            poll_two = process_two.poll()
-            if poll_two is None:
-                process_two.terminate()
-                process_two.wait(2)
-
-        os.chdir(self.t)
-        result = self.runner.invoke(
-            cli, [*CLI_LOG_OPTION, "delete", self.agent_name_one], standalone_mode=False
-        )
-        assert result.exit_code == 0
-        result = self.runner.invoke(
-            cli, [*CLI_LOG_OPTION, "delete", self.agent_name_two], standalone_mode=False
-        )
-        assert result.exit_code == 0
-
-    @classmethod
-    def teardown_class(cls):
-        """Teardowm the test."""
-        os.chdir(cls.cwd)
-        try:
-            shutil.rmtree(cls.t)
-        except (OSError, IOError):
-            pass
+        weather_station_aea_name = "my_weather_station"
+        weather_client_aea_name = "my_weather_client"
+
+        self.initialize_aea()
+        self.create_agents(weather_station_aea_name, weather_client_aea_name)
+
+        # prepare agent one (weather station)
+        weather_station_aea_dir_path = os.path.join(self.t, weather_station_aea_name)
+        os.chdir(weather_station_aea_dir_path)
+
+        self.add_item("connection", "fetchai/oef:0.2.0")
+        self.add_item("skill", "fetchai/weather_station:0.1.0")
+        self.disable_ledger_tx("fetchai", "skill", "weather_station")
```

Comment:
This method `disable_ledger_tx` needs to go from the test case, it is not generic at all (only works for one particular dotted path). Instead, use the config set functionality please.